test/e2e/federatedhpa_test.go

```diff
@@ -0,0 +1,219 @@
+			},
+		}
+		// pressure job always use Duplicated schedule type to prevent certain member cluster from missing it.
+		jobCPP = testhelper.NewClusterPropagationPolicy(pressureJobName, []policyv1alpha1.ResourceSelector{
+			{
+				APIVersion: pressureJob.APIVersion,
+				Kind:       pressureJob.Kind,
+				Name:       pressureJobName,
+			},
+		}, policyv1alpha1.Placement{
+			ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+				ReplicaSchedulingType: policyv1alpha1.ReplicaSchedulingTypeDuplicated,
+			},
+			ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+				ClusterNames: targetClusters,
+			},
+		})
+	})
+
+	ginkgo.JustBeforeEach(func() {
+		framework.CreateClusterPropagationPolicy(karmadaClient, cpp)
+		framework.CreateDeployment(kubeClient, deployment)
+		framework.CreateService(kubeClient, service)
+		framework.CreateFederatedHPA(karmadaClient, federatedHPA)
+
+		ginkgo.DeferCleanup(func() {
+			framework.RemoveFederatedHPA(karmadaClient, namespace, federatedHPAName)
+			framework.RemoveService(kubeClient, namespace, serviceName)
+			framework.RemoveDeployment(kubeClient, namespace, deploymentName)
+			framework.RemoveClusterPropagationPolicy(karmadaClient, cppName)
+			framework.WaitDeploymentDisappearOnClusters(targetClusters, namespace, deploymentName)
+			framework.WaitServiceDisappearOnClusters(targetClusters, namespace, serviceName)
+		})
+	})
+
+	// runPressureJob run a job to pressure the deployment to increase its cpu/mem.
+	var runPressureJob = func() {
+		framework.CreateClusterPropagationPolicy(karmadaClient, jobCPP)
+		framework.CreateJob(kubeClient, pressureJob)
+		framework.WaitJobPresentOnClustersFitWith(targetClusters, namespace, pressureJobName, func(_ *batchv1.Job) bool { return true })
+		ginkgo.DeferCleanup(func() {
+			framework.RemoveJob(kubeClient, namespace, pressureJobName)
+			framework.RemoveClusterPropagationPolicy(karmadaClient, pressureJobName)
+			framework.WaitJobDisappearOnClusters(targetClusters, namespace, pressureJobName)
+		})
+	}
+
+	// 1. duplicated scheduling
+	ginkgo.Context("FederatedHPA scale Deployment in Duplicated schedule type", func() {
+		ginkgo.BeforeEach(func() {
+			deployment.Spec.Replicas = pointer.Int32(1)
+		})
+
+		ginkgo.It("do scale when metrics of cpu/mem utilization up in Duplicated schedule type", func() {
+			ginkgo.By("step1: check initial replicas result should equal to cluster number", func() {
+				framework.WaitDeploymentStatus(kubeClient, deployment, clusterNum)
+			})
+
+			ginkgo.By("step2: pressure test the deployment of member clusters to increase the cpu/mem", runPressureJob)
+
+			ginkgo.By("step3: check final replicas result should greater than 1", func() {
+				framework.WaitDeploymentFitWith(kubeClient, namespace, deploymentName, func(deploy *appsv1.Deployment) bool {
+					return *deploy.Spec.Replicas > 1 && deploy.Status.Replicas > 1*clusterNum
```

Comment:
Is it enough to just determine spec.replicas?

Reply:
actually right